tests/Controllers/TimeEntryControllerTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Tests\Controllers;
+
+use App\Models\Task;
+use App\Models\TimeEntry;
+use App\Repositories\TimeEntryRepository;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Mockery\MockInterface;
+use Tests\TestCase;
+
+class TimeEntryControllerTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    /** @var MockInterface */
+    protected $timeEntryRepository;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->signInWithDefaultAdminUser();
+        $this->withHeaders(['X-Requested-With' => 'XMLHttpRequest']);
+    }
+
+    private function mockRepository()
+    {
+        $this->timeEntryRepository = \Mockery::mock(TimeEntryRepository::class);
+        app()->instance(TimeEntryRepository::class, $this->timeEntryRepository);
+    }
+
+    public function tearDown(): void
+    {
+        parent::tearDown();
+        \Mockery::close();
+    }
+
+    /** @test */
+    public function test_can_get_time_entry_details()
+    {
+        $this->mockRepository();
+
+        /** @var TimeEntry $timerEntry */
+        $timeEntry = factory(TimeEntry::class)->create();
+
+        $this->timeEntryRepository->shouldReceive('getTimeEntryDetail')
+            ->once()
+            ->with($timeEntry->id);
+
+        $response = $this->getJson("time-entries/$timeEntry->id/edit");
+
+        $this->assertSuccessMessageResponse($response, 'Time Entry retrieved successfully.');
```

Comment:
we need to assert time entry details here and all other places where a controller is returning a data in the response.